lib/raxx.ex

```diff
@@ -964,6 +964,59 @@ defmodule Raxx do
     |> set_header("content-type", "text/plain")
     |> set_body(reason_phrase)
   end
+
+  @doc """
+  TODO
+  """
+  def simplify_parts(parts) when is_list(parts) do
+    Enum.flat_map(parts, &simplify_part/1)
+  end
+
+  defp simplify_part(part = %Raxx.Data{}) do
+    [part]
+  end
+
+  defp simplify_part(part = %Raxx.Tail{}) do
+    [part]
+  end
+
+  defp simplify_part(response_headers = %Raxx.Response{body: true}) do
+    [response_headers]
+  end
+
+  defp simplify_part(response = %Raxx.Response{body: false}) do
+    # NOTE I'm not sure about this one
+    [response, Raxx.tail([])]
```

Comment:
If we assume a message always has a tail, i.e. `handle_tail` is always called. Then this looks the right way to do it.